### PR TITLE
slof_order: Add slof_order case for testing SLOF

### PR DESCRIPTION
**Pull request by YongxueHong**
slof_order case is that verify info of boot order.

ID: 1645056
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
Depends on #1531

**Comment by wangxianxian** [comment]
qemu.slof_order.network_disk_cdrom_network: ERROR: list index out of range (63.95 s)
qemu.slof_order.network_disk_network: ERROR: list index out of range (62.63 s)
qemu.slof_order.disk_network: ERROR: list index out of range (62.82 s)

The above 3 cases error when nicmodel=spapr-vlan, because the function verify_boot_device() in slof.py 
does not support spapr-vlan, I think it's better to support this.

**Comment by wangxianxian** [comment]
Ack

File: qemu/tests/slof_order.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    def _create_iso():
+        """Create iso with one file on it."""
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("creating iso.")
+            iso_name = params.get("cdrom_cd1")
+            iso = utils_misc.get_path(data_dir.get_data_dir(), iso_name)
+            process.system_output("dd if=/dev/urandom of=new bs=10M count=1",
+                                  300)
+            process.system_output("mkisofs -o %s new" % iso, 300)
+            process.system_output("rm -f new", 300)
```
Comment (lolyu) [inline qemu/tests/slof_order.py]: please use `os.remove` or `os.unlink`

File: qemu/tests/slof_order.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+                    'Pass: The device(%s@%s@%s) is the %d bootable device.' %
+                    (dev.parent, dev.child, dev.addr, index))
+            else:
+                test.fail(
+                    'Fail: The device(%s@%s@%s) is not the %d bootable device.'
+                    % (dev.parent, dev.child, dev.addr, index))
+
+    try:
+        _create_iso()
+        params['start_vm'] = 'yes'
```
Comment (lolyu) [inline qemu/tests/slof_order.py]: put it in the cfg pls.

Comment (lolyu) [reply qemu/tests/slof_order.py]: better to place LINE from 139 to 144  out this `try` block

File: qemu/tests/slof_order.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("creating iso.")
+            iso_name = params.get("cdrom_cd1")
+            iso = utils_misc.get_path(data_dir.get_data_dir(), iso_name)
+            process.system_output("dd if=/dev/urandom of=new bs=10M count=1",
+                                  300)
+            process.system_output("mkisofs -o %s new" % iso, 300)
+            process.system_output("rm -f new", 300)
+
+    def _cleanup_iso():
```
Comment (lolyu) [inline qemu/tests/slof_order.py]: you didn't remove the dd file.

File: qemu/tests/slof_order.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+                vm.send_key(key)
+        vm.send_key('ret')
+
+    def _create_iso():
+        """Create iso with one file on it."""
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("creating iso.")
+            iso_name = params.get("cdrom_cd1")
+            iso = utils_misc.get_path(data_dir.get_data_dir(), iso_name)
+            process.system_output("dd if=/dev/urandom of=new bs=10M count=1",
```
Comment (lolyu) [inline qemu/tests/slof_order.py]: dd file `new` better be placed in the `data_dir`

File: qemu/tests/slof_order.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    def _cleanup_iso():
+        """Removes created cdrom."""
+        if params.get("create_cd", 'no') == 'yes':
+            logging.info("cleaning up temp iso")
+            cdrom_test = utils_misc.get_path(
+                data_dir.get_data_dir(), params.get("cdrom_cd1"))
+            os.remove(cdrom_test)
+
+    def _verify_boot_once(content):
+        """Verify the specify device if booted once."""
```
Comment (lolyu) [inline qemu/tests/slof_order.py]: you could put `child_bus_type = 'scsi'` here and remove line 91 and 94

**Comment by YongxueHong** [comment]
@ xianwang @hereischen  help me review, thanks.

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by wangxianxian** [comment]
ACK

**Comment by YongxueHong** [comment]
thanks all.

**Pull request merged**